core/src/AzureMcp.Core/Areas/Server/Commands/Runtime/McpRuntime.cs

```diff
@@ -53,27 +52,11 @@ public McpRuntime(
     /// <param name="request">The request context containing the tool name and parameters.</param>
     /// <param name="cancellationToken">A token to monitor for cancellation requests.</param>
     /// <returns>A result containing the output of the tool invocation.</returns>
-    public async ValueTask<CallToolResult> CallToolHandler(RequestContext<CallToolRequestParams> request, CancellationToken cancellationToken)
+    public ValueTask<CallToolResult> CallToolHandler(RequestContext<CallToolRequestParams> request, CancellationToken cancellationToken)
     {
-        using var activity = _telemetry.StartActivity(ActivityName.ToolExecuted, request?.Server?.ClientInfo);
+        using var activity = _telemetry.StartActivity(ActivityName.ToolExecuted, request.Server.ClientInfo);
 
-        if (request?.Params == null)
-        {
-            var content = new TextContentBlock
-            {
-                Text = "Cannot call tools with null parameters.",
-            };
-
-            activity?.SetStatus(ActivityStatusCode.Error)?.AddTag(TagName.ErrorDetails, content.Text);
```

Comment:
I think we should continue to set our activity's status based on the "IsError" state of `_toolLoader.CallToolHandler`'s result

Reply:
Hmmm. Good idea!